### PR TITLE
feat: add hidden docs command

### LINKED ISSUE #1098: `rhoas docs` command to generate documentation in multiple formats
### Feature or problem description

I want to be able to generate documentation files locally in multiple formats. Currently, running the root command along with `GENERATE_DOCS=true` will generate documentation in Asciidoc format. Ideally we should be able to generate Markdown and Man pages also.

I mainly need to generate Man docs for official builds.

### Proposed solution

`rhoas docs -o ./docs --format=md,adoc,man`.

If needed, this can be enable only with `RHOAS_DEV=true` since users will probably not use this often or ever.

### PR DESCRIPTION
**Pull request by craicoverflow**
closes #1098 

this remains hidden from documentation and completions.

### Verification Steps

1. Run `rhoas docs -d ./docs/commands -f adoc` to generate documentation into the same folder as the old format.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: pkg/cmd/docs/docs.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		Hidden: true,
+		Args:   cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runCmd(cmd, &opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.format, "format", "f", "md", "Output format of the generated documentation. Valid options are: 'md' (markdown), 'adoc' (Asciidoc) and 'man'")
+	cmd.Flags().StringVarP(&opts.dir, "docs", "d", "./docs", "The directory to output the generated documentation files")
+
```
[the remark this answers is not in the document]
Comment (craicoverflow) [reply pkg/cmd/docs/docs.go:45]: true.

[the remark this answers is not in the document]
Comment (craicoverflow) [reply pkg/cmd/docs/docs.go:45]: fixed.

**Review by rkpattnaik780 — COMMENTED** [review]
Looks good.
Will it better to add validation and auto-completion to `--file-format`.

File: pkg/cmd/docs/docs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (craicoverflow) [reply pkg/cmd/docs/docs.go:45]: > Looks good.
> Will it better to add validation and auto-completion to `--file-format`.

No need, the command is hidden from completions anyway as it is intended for internal use. Can I get an approval?

**Comment by craicoverflow** [comment]
> Looks good.
> Will it better to add validation and auto-completion to `--file-format`.

No need, the command is hidden from completions anyway as it is intended for internal use. Can I get an approval?

**Review by rkpattnaik780 — APPROVED** [review]

**Pull request merged**